scripts/nlvr/generate_data_from_erm_model.py

```diff
@@ -0,0 +1,73 @@
+#! /usr/bin/env python
+
+# pylint: disable=invalid-name,wrong-import-position
+import sys
+import os
+import json
+import argparse
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir)))))
+
+from allennlp.data.dataset_readers import NlvrDatasetReader
+from allennlp.data.fields import MetadataField
+from allennlp.models.archival import load_archive
+from allennlp.semparse.worlds import NlvrWorld
+
+
+def make_data(input_file: str,
+              output_file: str,
+              archived_model_file: str,
+              max_num_decoded_sequences: int) -> None:
+    reader = NlvrDatasetReader(output_agendas=True)
+    model = load_archive(archived_model_file).model
+    num_outputs = 0
+    num_sentences = 0
+    with open(output_file, "w") as outfile:
+        for line in open(input_file):
+            num_sentences += 1
+            input_data = json.loads(line)
+            sentence = input_data["sentence"]
+            structured_representations = input_data["worlds"]
+            labels = input_data["labels"]
+            instance = reader.text_to_instance(sentence, structured_representations)
+            # Tweaking the instance to add an extra field to coerce the model to generate a k-best
+            # list. Setting k to 100 here, so that we can filter out the inconsistent ones later.
+            instance.fields["max_num_decoded_sequences"] = MetadataField(100)
+            outputs = model.forward_on_instance(instance, cuda_device=-1)
+            action_strings = outputs["best_action_strings"]
```

Comment:
Don't you already produce logical forms as output in `decode`?

Reply:
Yes, that's right. I'll also need the action strings because they will be written to the file, but I've changed this to take the logical forms from the outputs to avoid recomputing them.